src/EditorFeatures/Core.Wpf/InheritanceMargin/MarginGlyph/InheritanceMargin.xaml

```diff
@@ -0,0 +1,85 @@
+        <imaging:CrispImage x:Key="NonSharedIcon" x:Shared="False" Moniker="{Binding ImageMoniker}"/>
+
+        <!-- Style used to display a single target menu item -->
+        <Style x:Key="TargetMenuItemStyle" TargetType="{x:Type MenuItem}">
+            <Setter Property="Icon" Value="{StaticResource NonSharedIcon}"/>
+            <Setter Property="Header" Value="{Binding  DisplayName}"/>
+            <Setter Property="ToolTip" Value="{Binding ToolTip}"/>
+            <Setter Property="AutomationProperties.Name" Value="{Binding AutomationName}"/>
+            <Setter Property="IsCheckable" Value="False"/>
+            <EventSetter Event="Click" Handler="TargetMenuItem_OnHandler"/>
+        </Style>
+
+        <!-- The key is referenced from the code behind -->
+        <!-- Style used to display a single member's information, if click the button, it pops up a menu contains targets' information (one level menu)-->
+        <Style x:Key="SingleMemberContextMenuStyle" TargetType="{x:Type ContextMenu}">
+            <Setter Property="ItemContainerStyle" Value="{StaticResource TargetMenuItemStyle}"/>
+            <Setter Property="ItemsSource" Value="{Binding Targets}"/>
+            <Setter Property="StaysOpen" Value="True"/>
+            <Setter Property="Template">
+                <Setter.Value>
+                    <ControlTemplate TargetType="{x:Type ContextMenu}">
+                        <ScrollViewer VerticalScrollBarVisibility="Auto" Background="Transparent">
+                            <ItemsPresenter/>
+                        </ScrollViewer>
+                    </ControlTemplate>
+                </Setter.Value>
+            </Setter>
+        </Style>
+
+        <!-- Style used to display one member's information and pops up a submenu -->
+        <Style x:Key="MemberMenuItemStyle" TargetType="{x:Type MenuItem}">
+            <Setter Property="Icon" Value="{StaticResource NonSharedIcon}"/>
+            <Setter Property="Header" Value="{Binding DisplayName}"/>
+            <Setter Property="ToolTip" Value="{Binding ToolTip}"/>
+            <Setter Property="ItemsSource" Value="{Binding Targets}"/>
+            <Setter Property="ItemContainerStyle" Value="{StaticResource TargetMenuItemStyle}"/>
+            <Setter Property="IsCheckable" Value="False"/>
+            <Setter Property="AutomationProperties.Name" Value="{Binding AutomationName}"/>
+        </Style>
+
+        <!-- The key is referenced from the code behind -->
+        <!-- Style to show multiple members, when click the button it first pops a menu for all the members on this line (first level menu),
+        then each member has a second level of context menu, show its targerts -->
+        <Style x:Key="MultipleMembersContextMenuStyle" TargetType="ContextMenu">
+            <Setter Property="ItemContainerStyle" Value="{StaticResource MemberMenuItemStyle}"/>
```

Comment:
You can avoid some complexity in code by using conditional style setters in XAML. 

```xaml
<Style.Triggers>
    <DataTrigger Binding="{Binding ...}" Value="...">
        <Setter Property="ItemContainerStyle" Value="{StaticResource MemberMenuItemStyle}"/>
    </DataTrigger>
</Style.Triggers>
```

Reply:
Ah, this is tricky, so the margin is a button, and the context menu of the button is not a part of the Visual Tree. So it can't be binding in this way
<DataTrigger Binding = "{Binding {RelativeResouces AncesterType: {x:Type Button}}}"/>

Reply:
Or probably we need our own context menu type so that we could define a property called ``` ContainsMultipleMember ``` so that we could binding to this? I feel it becomes more complicated....

Reply:
> and the context menu of the button is not part of the visual tree

Is it not? Is that a requirement because of it being in the margin or some other constraint? 

> Or probably we need our own context menu type so that we could define a property called ContainsMultipleMember so that we could binding to this? I feel it becomes more complicated....

The data trigger doesn't have to be on the context menu, it just needs to be bindable. You can use the ViewModel for this as well. If the ContextMenu or Button have a VM that works.

Reply:
Actually after simplify the vm I found this is feasible. : ) Thanks Andrew